src/solidClientHelpers/settings.js

```diff
@@ -0,0 +1,130 @@
+  saveSolidDatasetAt,
+  addStringNoLocale,
+  createThing,
+  addUrl,
+  fetchResourceInfoWithAcl,
+} from "@inrupt/solid-client";
+import { title } from "rdf-namespaces/dist/dct";
+import { type } from "rdf-namespaces/dist/rdf";
+import { namespace } from "./index";
+
+async function getUserSettingsUrl(webId, { fetch }) {
+  const profileResource = await getSolidDataset(webId, { fetch });
+  const profile = getThing(profileResource, webId);
+  const settingsUrl = getUrl(profile, namespace.preferencesFile);
+  if (!settingsUrl) {
+    throw new Error(`Missing pointer to preferences file: ${settingsUrl}`);
+  }
+  return settingsUrl;
+}
+
+async function getOrCreatePodBrowserSettingsResource(
+  podBrowserSettingsUrl,
+  { fetch }
+) {
+  try {
+    await fetchResourceInfoWithAcl(podBrowserSettingsUrl, { fetch });
+    return podBrowserSettingsUrl;
+  } catch (error) {
+    if (!error.toString().match(/404/)) {
+      throw error;
+    }
+  }
+  // if error is 404, then create resource
+  const newSettings = createThing({ url: podBrowserSettingsUrl });
+  const settingsWithTitle = addStringNoLocale(
+    newSettings,
+    title,
+    "Pod Browser Preferences File"
+  );
+  const settingsWithType = addUrl(
+    settingsWithTitle,
+    type,
+    namespace.ConfigurationFile
+  );
+  await saveSolidDatasetAt(podBrowserSettingsUrl, settingsWithType, {
+    fetch,
+  });
+  return podBrowserSettingsUrl;
+}
+
+async function getOrCreatePodBrowserSettingsPointer(
+  userSettings,
+  userSettingsResource,
+  userSettingsUrl,
+  { fetch }
+) {
+  const podBrowserSettingsUrl = getUrl(
+    userSettings,
+    namespace.podBrowserPreferencesFile
+  );
+  if (podBrowserSettingsUrl) {
+    return Promise.resolve(podBrowserSettingsUrl);
+  }
+  // pointer does not already exist, so we create it
+  const newPodBrowserSettingsUrl = userSettingsUrl.replace(
+    /\/(\w+(.\w+)?)?$/g, // find the container that prefs.ttl resides in
+    "/podBrowserPrefs.ttl"
+  );
+  const updatedUserSettings = setUrl(
+    userSettings,
+    namespace.podBrowserPreferencesFile,
+    newPodBrowserSettingsUrl
+  );
+  const updatedUserSettingsResource = setThing(
+    userSettingsResource,
+    updatedUserSettings
+  );
+  await saveSolidDatasetAt(userSettingsUrl, updatedUserSettingsResource, {
+    fetch,
+  });
+  return Promise.resolve(newPodBrowserSettingsUrl);
+}
+
+async function getPodBrowserSettings(userSettingsUrl, session) {
+  const userSettingsResource = await getSolidDataset(userSettingsUrl, {
+    fetch: session.fetch,
+  });
+
+  const userSettings = getThing(userSettingsResource, userSettingsUrl);
+  const podBrowserSettingsUrl = await getOrCreatePodBrowserSettingsPointer(
+    userSettings,
+    userSettingsResource,
+    userSettingsUrl,
+    session
+  );
+  return getOrCreatePodBrowserSettingsResource(podBrowserSettingsUrl, session);
+}
+
+// eslint-disable-next-line import/prefer-default-export
+export async function getOrCreateSettings(webId, session) {
```

Comment:
Seems a little weird to me that `getOrCreateSettings` returns an iri instead of the settings dataset - I'd either rename this to `getOrCreateSettingsIri` or just return the dataset here.